File: client/src/app/models/userParams.ts
export class AnnouncementParams {
  pageNumber = 1;
  pageSize = 5;
  name: string;
  description: string;
  city: string;
  province: string;
  orderBy: string;
}


export class UserAnnouncementParams {
  pageNumber = 1;
  pageSize = 5;
  isActive = true;
  orderBy: string;
}

export class MemberParams {
  pageNumber = 1;
  pageSize = 5;
  userName: string;
  firstName: string;
  lastName: string;
  gender: string;
  role: string;
  orderBy: string;
}

export class AdminAnnouncementParams {
  pageNumber = 1;
  pageSize = 5;
}

File: API/Helpers/UserParams.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string userName { get; set; }
        public string firstName { get; set; }
        public string lastName { get; set; }
        public string gender { get; set; }
        public string Role { get; set; }
        public string OrderBy { get; set; } = "userName";
    }
}
